### ISSUE TITLE
Orca: AttributeError: 'PyTorchPySparkEstimator' object has no attribute 'port' in est.shutdown() with cluster_mode='yarn-client'

### ISSUE THREAD

**Issue by zpeng1898** [intel-analytics/BigDL#7467]
When I run NCF example `pytorch_train_spark_dataframe.py` with `cluster_mode="yarn-client"`, an error occurs in `estimator.shutdown()` showing `AttributeError: 'PyTorchPySparkEstimator' object has no attribute 'port'`. However, no error occurs with `cluster_mode="local"`.

- Reproduction:
```
from bigdl.orca import init_orca_context
from bigdl.orca.learn.pytorch import Estimator

sc = init_orca_context(cluster_mode="yarn-client",
                                   cores=4, memory="10g", num_nodes=2,
                                   extra_python_lib=extra_python_lib)
est = Estimator.from_torch(model=model_creator,
                           optimizer=optimizer_creator,
                           loss=loss,
                           scheduler_creator=scheduler_creator,
                           metrics=[Accuracy(), Precision(), Recall()],
                           config=config,
                           backend=args.backend,
                           use_tqdm=True,
                           workers_per_node=args.workers_per_node)
est.shutdown()
stop_orca_context()
```

- error message
```
[partition = 1, ip = 172.16.0.118] [2023-02-08 14:28:01] INFO     Connected log server on 172.16.0.107:59015
[partition = 0, ip = 172.16.0.118] [2023-02-08 14:28:01] INFO     Connected log server on 172.16.0.107:59015
[partition = 1, ip = 172.16.0.118] Data size on worker:  49923
[partition = 0, ip = 172.16.0.118] Data size on worker:  50025
Evaluation results:                                                             
num_samples: 99948
Accuracy: 0.8572947978973389
Precision: 0.7898339629173279
Recall: 0.39525434374809265
val_loss: 0.32110947912749455
Traceback (most recent call last):                                              
  File "pytorch_train_spark_dataframe.py", line 126, in <module>
    est.shutdown()
  File "/home/kai/anaconda3/envs/py378/lib/python3.7/site-packages/bigdl/orca/learn/pytorch/pytorch_pyspark_estimator.py", line 705, in shutdown
    stop_log_server(self.log_server_thread, self.ip, self.port)
AttributeError: 'PyTorchPySparkEstimator' object has no attribute 'port'
Stopping orca context
```

**Comment by hkvision** [comment]
@lalalapotter @sgwhat Please take a look.

**Comment by sgwhat** [comment]
Solved before with PR https://github.com/intel-analytics/BigDL/pull/5904, but seems it only was updated on `branch-2.1` LOL, we will create another pr to sync the change.

**Comment by hkvision** [comment]
This should be fixed in #7471

**Comment by hkvision** [comment]
Tested using the latest version and succeeded.
